homeassistant/components/ws66i/media_player.py

```diff
@@ -0,0 +1,214 @@
+    | SUPPORT_SELECT_SOURCE
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+@core.callback
+def _get_sources_from_dict(data):
+    sources_config = data[CONF_SOURCES]
+
+    # Dict index to custom name
+    source_id_name = {int(index): name for index, name in sources_config.items()}
+
+    # Dict custom name to index
+    source_name_id = {v: k for k, v in source_id_name.items()}
+
+    # List of custom names
+    source_names = sorted(source_name_id.keys(), key=lambda v: source_name_id[v])
+
+    return [source_id_name, source_name_id, source_names]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the WS66i 6-zone amplifier platform."""
+    ip_addr = config_entry.data[CONF_IP_ADDRESS]
+    ws66i = hass.data[DOMAIN][config_entry.entry_id][WS66I_OBJECT]
+    sources = _get_sources_from_dict(config_entry.options)
+
+    # Build the entities that control each of the zones.
+    # Zones 11 - 16 are the master amp
+    # Zones 21,31 - 26,36 are the daisy-chained amps
+    entities = []
+    for i in range(1, 4):
+
+        if i > 1:
+            # Don't add entities that aren't present
+            status = await hass.async_add_executor_job(ws66i.zone_status, (i * 10 + 1))
+            if status is None:
+                break
+
+        _LOGGER.info("Detected amp %d at ip %s", i, ip_addr)
+        for j in range(1, 7):
+            zone_id = (i * 10) + j
+            entities.append(Ws66iZone(ws66i, sources, config_entry.entry_id, zone_id))
+
+    async_add_entities(entities)
+
+    # Set up services
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_SNAPSHOT,
+        {},
+        "snapshot",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_RESTORE,
+        {},
+        "restore",
+    )
+
+
+class Ws66iZone(MediaPlayerEntity):
+    """Representation of a WS66i amplifier zone."""
+
+    def __init__(self, ws66i, sources, namespace, zone_id):
+        """Initialize new zone."""
+        self._ws66i = ws66i
+        self._zone_id = zone_id
+        # dict source_id -> source name
+        self._source_id_name = sources[0]
+        # dict source name -> source_id
+        self._source_name_id = sources[1]
+        # ordered list of all source names
+        self._attr_source_list = sources[2]
+        self._attr_unique_id = f"{namespace}_{self._zone_id}"
+        self._attr_name = f"Zone {self._zone_id}"
+        self._attr_supported_features = SUPPORT_WS66I
+        self._attr_available = True
+        self._attr_should_poll = True
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Soundavo",
+            "model": "WS66i 6-Zone Amplifier",
+        }
+
+        self._snapshot = None
+        self._state = None
+        self._volume = None
+        self._source = None
+        self._mute = None
```

Comment:
```suggestion
        self._attr_is_volume_muted = None
```

Reply:
There is a complete list of ones that can be converted here

https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/__init__.py#L400

Reply:
Thanks. This is useful.